server/index.js

```diff
@@ -1,36 +1,18 @@
-dotenv.config();
 import express from 'express';
-import dotenv from 'dotenv';
-import cookieParser from 'cookie-parser';
-
-import { notFound, errorHandler } from './middleware/errorMiddleware.js';
-import connectDB from './database/connection.js';
-import userRoutes from './routes/userRoutes.js'
-import Product from './models/Product.js';
-
-const port = process.env.PORT || 4000;
-
+import dotenv from 'dotenv'
+import routes from './routes/routes.js'
+import db from './database/connection.js';
+dotenv.config();
 connectDB();
```

Comment:
This returns a Promise, which is not handled

Reply:
This will crash the server, if in case the DB fails to connect, with the error `unhandled promise rejection`

Reply:
The promise is still not handled, here's a hint use `try` `catch` block

Reply:
used try and catch to handle it